extensions/amp-lightbox-gallery/0.1/test/test-amp-lightbox-gallery.js

```diff
@@ -14,212 +14,38 @@
-    viewer.setAttribute('layout', 'nodisplay');
     installLightboxManager(win);
-    doc.body.appendChild(viewer);
-    return viewer.build()
-        .then(() => viewer.layoutCallback())
-        .then(() => {
-          const impl = viewer.implementation_;
-          // stub vsync and resource function
-          sandbox.stub(impl.vsync_, 'mutate').callsFake(callback => {
-            callback();
-          });
-          sandbox.stub(impl.vsync_, 'mutatePromise').callsFake(callback => {
-            callback();
-            return Promise.resolve();
-          });
-          sandbox.stub(impl.resources_, 'requireLayout').callsFake(() => {
-            return Promise.resolve();
-          });
-        })
-        .then(() => viewer);
-  }
-
-  // TODO (cathyzhu): rewrite these tests after finalizing lightbox API.
-  describe.skip('with manual lightboxing', function() {
-    runTests(/*autoLightbox*/false);
+    installLightboxGallery(win);
+    gallery = doc.getElementById(TAG);
   });
 
-  describe.skip('with auto lightboxing', function() {
-    runTests(/*autoLightbox*/true);
-  });
-
-  function runTests(autoLightbox) {
-    it('should build on open', () => {
-      let viewer = null;
-      return getAmpLightboxViewer(autoLightbox).then(v => {
-        viewer = v;
-        const impl = viewer.implementation_;
-        expect(viewer.style.display).to.equal('none');
-        return impl.open_(item1);
-      }).then(() => {
-        const container = viewer.querySelector('.i-amphtml-lbv');
-        expect(container).to.exist;
-
-        const mask = viewer.querySelector('.i-amphtml-lbv-mask');
-        expect(mask).to.exist;
-
-        const topBar = viewer.querySelector('.i-amphtml-lbv-top-bar');
-        expect(topBar).to.exist;
-
-        const btns = viewer.querySelectorAll(
-            '.i-amphtml-lbv-top-bar > [role=button]');
-        expect(btns.length).to.equal(3);
-        expect(btns[0].className).to.equal('amp-lbv-button-close');
-        expect(btns[1].className).to.equal('amp-lbv-button-gallery');
-        expect(btns[2].className).to.equal('amp-lbv-button-slide');
-
-        const descriptionBox = viewer.querySelector('.i-amphtml-lbv-desc-box');
-        expect(descriptionBox).to.exist;
-
-        const descriptionTextArea = viewer.querySelector(
-            '.i-amphtml-lbv-desc-text');
-        expect(descriptionTextArea).to.exist;
-
-        const carousel = viewer.querySelector('amp-carousel');
-        expect(carousel).to.exist;
+  describe('basic functionality', function() {
+    this.timeout(5000);
+    it('should contain a container on build', done => {
+      gallery.build().then(() => {
+        const container = doc.getElementsByClassName('i-amphtml-lbg');
+        expect(container.length).to.equal(1);
+        expect(container[0].tagName).to.equal('DIV');
+        done();
       });
     });
 
-    it('should make lightbox viewer visible on activate', () => {
```

Comment:
Could you help me understand why you are removing so many tests? Is this behavior no longer controlled by the gallery itself?

Reply:
Yup. I removed all unit tests because they were originally skipped anyway. In the early stages of amp-lightbox-gallery we completely refactored it to make these unit tests no longer applicable. I removed the test file as an attempt to fix the build failing when no tests are run. However, that didn't fix the build, so I wrote one new unit test to make the build pass. 

It makes very little sense to unit test `amp-lightbox-gallery` as it currently is, so I'm investing more in integration tests instead, e.g. per https://github.com/ampproject/amphtml/pull/15932.